scipy/sparse/linalg/matfuncs.py

```diff
@@ -605,6 +605,11 @@ def _expm(A, use_exact_onenorm):
     if len(A.shape) != 2 or A.shape[0] != A.shape[1]:
         raise ValueError('expected a square matrix')
 
+    # gracefully handle size-0 input,
+    # carefully handling sparse scenario
+    if np.asarray(A).size == 0:
```

Comment:
Check instead `A.shape == (0, 0)`, because `A` should not be casted to numpy array here, and return `A.__class__([[]])` so that the return value is a new array.

Reply:
That fails with:

```
        # gracefully handle size-0 input,
        # carefully handling sparse scenario
        if A.shape == (0, 0):
>           return A.__class__([[]])
E           TypeError: 'list' object cannot be interpreted as an integer

A          = array([], shape=(0, 0), dtype=float64)
use_exact_onenorm = 'auto'

scipy/sparse/linalg/matfuncs.py:611: TypeError

```

Reply:
Ok, then it proably has to be how it's a few lines below:
```
out = np.zeros([0, 0])
if isspmatrix(A) or is_pydata_spmatrix(A):
    return A.__class__(out)
return out
```
It's repetition, but I wouldn't start refactoring for this.

Reply:
That worked, thanks